diff --git a/monero_libwallet2_api_c/monero_libwallet2_api_c.exp b/monero_libwallet2_api_c/monero_libwallet2_api_c.exp
index 1aa2f26..7f3e164 100644
--- a/monero_libwallet2_api_c/monero_libwallet2_api_c.exp
+++ b/monero_libwallet2_api_c/monero_libwallet2_api_c.exp
@@ -282,9 +282,6 @@ _MONERO_WalletManager_blockchainTargetHeight
 _MONERO_WalletManager_networkDifficulty
 _MONERO_WalletManager_miningHashRate
 _MONERO_WalletManager_blockTarget
-_MONERO_WalletManager_isMining
-_MONERO_WalletManager_startMining
-_MONERO_WalletManager_stopMining
 _MONERO_WalletManager_resolveOpenAlias
 _MONERO_WalletManager_setProxy
 _MONERO_WalletManagerFactory_getWalletManager
diff --git a/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.cpp b/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.cpp
index c58fbec..5239e74 100644
--- a/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.cpp
+++ b/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.cpp
@@ -2387,27 +2387,6 @@ uint64_t MONERO_WalletManager_blockTarget(void* wm_ptr) {
     return wm->blockTarget();
     DEBUG_END()
 }
-//     virtual bool isMining() = 0;
-bool MONERO_WalletManager_isMining(void* wm_ptr) {
-    DEBUG_START()
-    Monero::WalletManager *wm = reinterpret_cast<Monero::WalletManager*>(wm_ptr);
-    return wm->isMining();
-    DEBUG_END()
-}
-//     virtual bool startMining(const std::string &address, uint32_t threads = 1, bool background_mining = false, bool ignore_battery = true) = 0;
-bool MONERO_WalletManager_startMining(void* wm_ptr, const char* address, uint32_t threads, bool backgroundMining, bool ignoreBattery) {
-    DEBUG_START()
-    Monero::WalletManager *wm = reinterpret_cast<Monero::WalletManager*>(wm_ptr);
-    return wm->startMining(std::string(address), threads, backgroundMining, ignoreBattery);
-    DEBUG_END()
-}
-//     virtual bool stopMining() = 0;
-bool MONERO_WalletManager_stopMining(void* wm_ptr, const char* address) {
-    DEBUG_START()
-    Monero::WalletManager *wm = reinterpret_cast<Monero::WalletManager*>(wm_ptr);
-    return wm->stopMining();
-    DEBUG_END()
-}
 //     virtual std::string resolveOpenAlias(const std::string &address, bool &dnssec_valid) const = 0;
 const char* MONERO_WalletManager_resolveOpenAlias(void* wm_ptr, const char* address, bool dnssec_valid) {
     DEBUG_START()
diff --git a/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.h b/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.h
index ff28d68..e9d6078 100644
--- a/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.h
+++ b/monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.h
@@ -966,12 +966,6 @@ extern ADDAPI uint64_t MONERO_WalletManager_networkDifficulty(void* wm_ptr);
 extern ADDAPI double MONERO_WalletManager_miningHashRate(void* wm_ptr);
 //     virtual uint64_t blockTarget() = 0;
 extern ADDAPI uint64_t MONERO_WalletManager_blockTarget(void* wm_ptr);
-//     virtual bool isMining() = 0;
-extern ADDAPI bool MONERO_WalletManager_isMining(void* wm_ptr);
-//     virtual bool startMining(const std::string &address, uint32_t threads = 1, bool background_mining = false, bool ignore_battery = true) = 0;
-extern ADDAPI bool MONERO_WalletManager_startMining(void* wm_ptr, const char* address, uint32_t threads, bool backgroundMining, bool ignoreBattery);
-//     virtual bool stopMining() = 0;
-extern ADDAPI bool MONERO_WalletManager_stopMining(void* wm_ptr, const char* address);
 //     virtual std::string resolveOpenAlias(const std::string &address, bool &dnssec_valid) const = 0;
 extern ADDAPI const char* MONERO_WalletManager_resolveOpenAlias(void* wm_ptr, const char* address, bool dnssec_valid);
 //     static std::tuple<bool, std::string, std::string, std::string, std::string> checkUpdates(
